lib/redd/objects/subreddit.rb

```diff
@@ -41,7 +41,7 @@ def submit(
           resubmit: resubmit, sendreplies: sendreplies
         }
 
-        params << {captcha: captcha, iden: identifier} if captcha
+        params.merge!(captcha: captcha, iden: identifier) if captcha
```

Comment:
@Fustrate 
I'm not sure but maybe would be better to check presence of `identifier` too

```ruby
params.merge!(captcha: captcha, iden: identifier) if captcha && identifier
```